lib/jekyll/filters.rb

```diff
@@ -389,7 +389,11 @@ def groupable?(element)
     private
     def item_property(item, property)
       if item.respond_to?(:to_liquid)
-        item.to_liquid[property.to_s]
+        res = item.to_liquid
+        property.to_s.split(".").each do |attr|
+          res = res[attr]
+        end
+        res
```

Comment:
I'm would use `subvalue` and `attribute` instead of `res` and `prop` but I can't because the line can't exceed 90 characters.

I also can't make it multine like this because it's not allowed in the rules.
```
        property.to_s.split(".").reduce(item.to_liquid) { |val, attr|
          val[attr]
        }
```